doc/source/ScriptingAPI/DictionarySyntax/index.md

```diff
@@ -0,0 +1,65 @@
+# Dictionary Syntax #
+
+Gaffer scripts use Python's dictionary syntax and notation to define the node graph. As you add to and manipulate the node graph, the dictionary builds a compendium of instances of node and plug classes. Each node is a dictionary key, with a node class as its value. Each plug is a sub-key of a node's key, with a plug class as its value. The following pseudocode shows a rough approximation of a node graph's dictionary:
+
+```python
+script = {
```

Comment:
I'm open to suggestions for more accurate pseudocode for this. Might be better represented by a diagram.

Reply:
Hmm, putting square brackets around the keys is definitely confusing.  There does seem to be a bit of a problem in presenting things this way, in that it would tend to be interpreted as "the script variable in the script editor points to something that is actually a Python dictionary ( which is false )", when what you mean is that "it can be accessed like a Python dictionary".  If you're going to put code here, maybe it would be better to show code for accessing nodes and plug in the script, rather than making it looking like you're defining the script?

Reply:
I wanted to demonstrate the hierarchy and syntax at once, similar to demonstrating a high-level C-style class hierarchy:

```C++
class foo {
    class bar {
        class becue {}
    }
    class boo {}
}
```

As far as I've read, you can't display a dictionary in code like this. By the sounds of things, a diagram would be the less confusing option.